components/clipboard-button/index.js

```diff
@@ -78,13 +79,15 @@ class ClipboardButton extends Component {
 		// Disable reason: Exclude from spread props passed to Button
 		// eslint-disable-next-line no-unused-vars
 		const { className, children, onCopy, onFinishCopy, text, ...buttonProps } = this.props;
+		const { icon } = buttonProps;
 		const classes = classnames( 'components-clipboard-button', className );
+		const ComponentToUse = icon ? IconButton : Button;
 
 		return (
 			<span ref={ this.bindContainer }>
-				<IconButton { ...buttonProps } icon="admin-links" className={ classes }>
+				<ComponentToUse { ...buttonProps } icon={ icon } className={ classes }>
```

Comment:
Assigning `icon` explicitly is technically redundant. In practice, this would be transformed to:

```js
createElement( ComponentToUse, { children: '...', icon: 'admin-links' }, { icon: 'admin-links', className: '...' } )
```

...since `icon` is already included in `buttonProps`.